plugin/parser/engine/pg/convert.go

```diff
@@ -765,3 +769,92 @@ func convertToTableType(relationType pgquery.ObjectType) (ast.TableType, error)
 		return ast.TableTypeUnknown, parser.NewConvertErrorf("expected TABLE or VIEW but found %s", relationType)
 	}
 }
+
+func filterPgCatalogPrefix(tp *pgquery.TypeName) *pgquery.TypeName {
+	// The built-in data type may have the "pg_catalog" prefix.
+	if len(tp.Names) > 0 {
+		if first, ok := tp.Names[0].Node.(*pgquery.Node_String_); ok && first.String_.Str == "pg_catalog" {
+			tp.Names = tp.Names[1:]
+		}
+	}
+	return tp
+}
+
+func convertDataType(tp *pgquery.TypeName) ast.DataType {
+	tp = filterPgCatalogPrefix(tp)
+	switch len(tp.Names) {
+	case 1:
+		name, ok := tp.Names[0].Node.(*pgquery.Node_String_)
+		if !ok {
+			return &ast.UnconvertedDataType{}
+		}
+		s := name.String_.Str
+		switch {
+		case strings.HasPrefix(s, "int"):
+			size, err := strconv.Atoi(s[3:])
+			if err != nil {
+				return &ast.UnconvertedDataType{}
+			}
+			return &ast.Integer{Size: size}
+		case strings.HasPrefix(s, "float"):
+			size, err := strconv.Atoi(s[5:])
+			if err != nil {
+				return &ast.UnconvertedDataType{}
+			}
+			return &ast.Float{Size: size}
+		case s == "serial":
```

Comment:
I thought the serial will be converted to the integer type with an additional sequence for increments.

create table hello(id serial);

# \d hello
                            Table "public.hello"
 Column |  Type   | Collation | Nullable |              Default              
--------+---------+-----------+----------+-----------------------------------
 id     | integer |           | not null | nextval('hello_id_seq'::regclass)

Reply:
Same question

Reply:
Now we skip all default constraints.
I'll add the default constraint for the serial type when we need it.